modules/user-interface-configuration/src/main/java/org/opencastproject/uiconfig/UIConfigRest.java

```diff
@@ -69,48 +74,65 @@
             + "<a href=\"https://github.com/opencast/opencast/issues\">Opencast Issue Tracker</a>"
     }
 )
+@Component(
+    property = {
+        "service.description=UI Config REST Endpoint",
+        "opencast.service.type=org.opencastproject.uiconfig",
+        "opencast.service.path=/ui/config"
+    },
+    immediate = true,
+    service = UIConfigRest.class
+)
 public class UIConfigRest {
   /** The logger */
   private static final Logger logger = LoggerFactory.getLogger(UIConfigRest.class);
 
   /** Configuration key for the ui config folder */
   static final String UI_CONFIG_FOLDER_PROPERTY = "org.opencastproject.uiconfig.folder";
+  static final String X_ACCEL_REDIRECT_PROPERTY = "x.accel.redirect";
 
   /** Default Path for the ui configuration folder (relative to ${karaf.etc}) */
   private static final String UI_CONFIG_FOLDER_DEFAULT = "ui-config";
 
   /** The currently used path to the configuration folder */
   private String uiConfigFolder = "";
+  private String xAccelRedirect = null;
 
   /** The used SecurityService */
   private SecurityService securityService;
 
+  @Reference
   protected void setSecurityService(SecurityService securityService) {
     this.securityService = securityService;
   }
 
-
   /**
    * OSGI callback for activating this component
    *
    * @param cc
    *          the osgi component context
    */
-  public void activate(ComponentContext cc) throws ConfigurationException {
-    uiConfigFolder = cc.getBundleContext().getProperty(UI_CONFIG_FOLDER_PROPERTY);
-
-    if (StringUtils.isEmpty(uiConfigFolder)) {
-      String karafetc = cc.getBundleContext().getProperty("karaf.etc");
-
-      if (StringUtils.isBlank(karafetc)) {
-        throw new ConfigurationException(UI_CONFIG_FOLDER_PROPERTY + " not set and unable to"
-                                         + " fall back to default location based on ${karaf.etc}");
+  @Activate
+  @Modified
+  public void activate(BundleContext context, Map<String, String> properties)
+          throws ConfigurationException, IOException {
+    uiConfigFolder = properties.get(UI_CONFIG_FOLDER_PROPERTY);
+    logger.debug("UI configuration folder set to '{}'", uiConfigFolder);
```

Comment:
Do we really need this if we always log the canonical path further down? If we do, maybe we should differentiate the logging message a bit.

Reply:
Here you see what the actual configured value is while magic is involved in the other log further down. They are both debug level logs, and you wouldn't normally see them. But I don't care that much, and I can remove either one if you want me to.

Reply:
```suggestion
    logger.debug("UI configuration folder configured as '{}'", uiConfigFolder);
```
?